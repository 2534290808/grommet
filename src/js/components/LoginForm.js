// (C) Copyright 2014-2015 Hewlett-Packard Development Company, L.P.

var React = require('react');

var CLASS_NAME = "login-form";

var LoginForm = React.createClass({

  propTypes: {
    logo: React.PropTypes.node,
    title: React.PropTypes.string,
    rememberMe: React.PropTypes.bool,
    forgotPassword: React.PropTypes.node,
    errors: React.PropTypes.arrayOf(React.PropTypes.string),
    onSubmit: React.PropTypes.func
  },

  _onSubmit: function (event) {
    event.preventDefault();
    var username = this.refs.username.getDOMNode().value.trim();
    var password = this.refs.password.getDOMNode().value.trim();
    if (this.props.onSubmit) {
      this.props.onSubmit({username: username, password: password});
    }
  },

  getDefaultProps: function () {
    return ({
      errors: []
    });
  },

  componentDidMount: function() {
    this.refs.username.getDOMNode().focus();
  },

  render: function() {
    var classes = [CLASS_NAME];

    var errors = this.props.errors.map(function (error, index) {
      return (<div key={index} className={CLASS_NAME + "__error"}>{error}</div>);
    });

    var logo = null;
    if (this.props.logo) {
      logo = (
        <div className={CLASS_NAME + "__logo"}>
          {this.props.logo}
        </div>
      );
    }

    var title = null;
    if (this.props.title) {
      title = (
        <h1 className={CLASS_NAME + "__title"}>
          {this.props.title}
        </h1>
      );
    }

    var footer = null;
    if (this.props.rememberMe || this.props.forgotPassword) {
      var rememberMe = null;
      if (this.props.rememberMe) {
        rememberMe = (
          <span className={CLASS_NAME + "__remember-me"}>
            <input id="remember-me" type="checkbox" />
            <label htmlFor="remember-me" className="checkbox">Remember me</label>
          </span>
        );
      }
      footer = (
        <div className={CLASS_NAME + "__footer"}>
          {rememberMe}
          {this.props.forgotPassword}
        </div>
      );
    }

    return (
      <form className={classes.join(' ')} onSubmit={this._onSubmit}>
        {logo}
        {title}
        <fieldset>
          <label>Username</label>
          <input className={CLASS_NAME + "__username"} ref="username" />
          <label>Password</label>
          <input type="password" className={CLASS_NAME + "__password"} ref="password" />
        </fieldset>
        {errors}
        <input type="submit" className={CLASS_NAME + "__submit primary call-to-action"} value={'Log in'} />
        {footer}
      </form>
    );
  }

});

module.exports = LoginForm;
