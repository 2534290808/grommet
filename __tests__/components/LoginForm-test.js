// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import LoginForm from '../../src/js/components/LoginForm';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('LoginForm', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <LoginForm onSubmit={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
