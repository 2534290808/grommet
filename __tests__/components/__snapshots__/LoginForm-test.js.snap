exports[`LoginForm has correct default options 1`] = `
<form
  className="grommetux-form grommetux-form--pad-medium grommetux-login-form"
  onSubmit={[Function bound _onSubmit]}>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined} />
  <fieldset>
    <div
      className="grommetux-form-field grommetux-form-field--text grommetux-form-field--size-medium"
      onClick={[Function bound _onClick]}>
      <label
        className="grommetux-form-field__label"
        htmlFor="username">
        <span>
          Email
        </span>
      </label>
      <span
        className="grommetux-form-field__contents">
        <input
          onChange={[Function bound _onUsernameChange]}
          type="email"
          value="" />
      </span>
    </div>
    <div
      className="grommetux-form-field grommetux-form-field--text grommetux-form-field--size-medium"
      onClick={[Function bound _onClick]}>
      <label
        className="grommetux-form-field__label"
        htmlFor="password">
        <span>
          Password
        </span>
      </label>
      <span
        className="grommetux-form-field__contents">
        <input
          onChange={[Function bound _onPasswordChange]}
          type="password"
          value="" />
      </span>
    </div>
  </fieldset>
  <footer
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-stretch grommetux-box--pad-vertical-none grommetux-box--pad-between-medium grommetux-footer grommetux-footer--small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <button
      aria-label={undefined}
      className="grommetux-button grommetux-button--primary grommetux-button--fill"
      disabled={false}
      href={undefined}
      onBlur={[Function bound _onBlur]}
      onClick={[Function bound _onSubmit]}
      onFocus={[Function bound _onFocus]}
      onMouseDown={[Function bound _onMouseDown]}
      onMouseUp={[Function bound bound _onMouseDown]}
      type="submit">
      <span
        className="grommetux-button__label">
        <span>
          Log In
        </span>
      </span>
    </button>
  </footer>
</form>
`;
